File: 2024/day1/day1.py
## DAY 1

list1 = []
list2 = []


def readFile():
    with open('puzzleInput.txt', 'r') as file:
        for line in file:
            num1, num2 = map(int, line.split())
            list1.append(num1)
            list2.append(num2)




def compare(list1, list2):
    longest = max(len(list1), len(list2))
    result = []
    i = 0
    while( i < longest):
        if(list1[i] >=  list2[i]):
            result.append(list1[i] - list2[i])
        else:
            result.append(list2[i] - list1[i])
        i = i + 1
    
    return result



def run():
    readFile()
    list1.sort()
    list2.sort()
    result = compare(list1,list2)
    return sum(result)

File: 2024/day1/day1.hs
module Day1 where 

import Data.List(sort)
import System.IO(readFile)

run :: IO () 
run = do
    content <- readFile "puzzleInput.txt"
    let rows = lines content                      
        columns = map (words) rows                
        column1 = map (!! 0) columns              
        column2 = map (!! 1) columns              
    let column1Nums = map read column1 :: [Int]
        column2Nums = map read column2 :: [Int]
    print (prep column1Nums column2Nums)

prep :: [Int] -> [Int] -> Int
prep xs ys =  sum $ comp (sort xs) (sort ys) 

comp :: [Int] -> [Int] -> [Int]
comp _ []       = []
comp (x:xs) (y:ys) 
    | x >= y    = (x - y) : comp xs ys
    | otherwise = (y - x) : comp xs ys

